include/zephyr/newlib_types.h

```diff
@@ -0,0 +1,42 @@
+
+#undef  __INT8_TYPE__
```

Comment:
What it the idea behind of this? Now we are basically taking all the defines from the GCC internals, the ones that are defined by the compiler and we can be sure that this values are for the target machine, and we are redefining it to our own types? Why?

Reply:
> What it the idea behind of this? Now we are basically taking all the defines from the GCC internals, the ones that are defined by the compiler and we can be sure that this values are for the target machine, and we are redefining it to our own types? Why?

In Zephyr we already define target machine types in include/zephyr/types.h that we using.  The intent of this change is to have newlib match how we define those types so things are in sync.